pkg/configuration/config.go

```diff
@@ -266,12 +295,44 @@ func Set(scope string, options map[string]string) {
 		utils.HandleError(err, fmt.Sprintf("Invalid scope: %s", scope))
 	}
 
+	config := configContents.Scoped[normalizedScope]
+	previousToken := config.Token
+
 	for key, value := range options {
 		if !IsValidConfigOption(key) {
 			utils.HandleError(errors.New("invalid option "+key), "")
 		}
 
-		config := configContents.Scoped[normalizedScope]
+		if key == models.ConfigToken.String() {
+			utils.LogDebug(fmt.Sprintf("Saving %s to system keyring", key))
+			uuid, err := uuid.NewRandom()
+			if err != nil {
+				utils.HandleError(err, "Unable to generate random UUID")
+			}
+			id := controllers.GenerateKeyringID(uuid.String())
+
+			if controllerError := controllers.SetKeyring(id, value); !controllerError.IsNil() {
```

Comment:
Can you add some comment to the different path here? It looks like this is some backward compatibility logic, but not sure

Reply:
Can you clarify what you mean here, I don't follow

Reply:
Nvm, ignore